src/java/org/apache/cassandra/index/sai/plan/VectorTopKProcessor.java

```diff
@@ -147,6 +201,64 @@ public <U extends Unfiltered, R extends BaseRowIterator<U>, P extends BasePartit
         return new InMemoryUnfilteredPartitionIterator(command, unfilteredByPartition);
     }
 
+    private <U extends Unfiltered> List<Triple<PartitionInfo, Row, Float>> processPart(BaseRowIterator<U> part,
+                                                    SortedMap<PartitionInfo, TreeSet<Unfiltered>> unfilteredByPartition,
+                                                    AbstractQueue<Triple<PartitionInfo, Row, Float>> topK,
+                                                    boolean isParallel)
+    {
+        // compute key and static row score once per partition
+        DecoratedKey key = part.partitionKey();
+        Row staticRow = part.staticRow();
+        final PartitionInfo partitionInfo = PartitionInfo.create(part);
+        final float keyAndStaticScore = getScoreForRow(key, staticRow);
+
+        List<Triple<PartitionInfo, Row, Float>> res = new LinkedList<>();
+        while (part.hasNext())
+        {
+            Unfiltered unfiltered = part.next();
+            // Always include tombstones for coordinator. It relies on ReadCommand#withMetricsRecording to throw
+            // TombstoneOverwhelmingException to prevent OOM.
+            if (!unfiltered.isRow())
+            {
+                if (isParallel)
+                {
+                    synchronized (unfilteredByPartition)
+                    {
+                        addUnfiltered(unfilteredByPartition, partitionInfo, unfiltered);
+                    }
+                }
+                else
+                {
+                    addUnfiltered(unfilteredByPartition, partitionInfo, unfiltered);
+                }
+                continue;
+            }
+
+            Row row = (Row) unfiltered;
+            float rowScore = getScoreForRow(null, row);
+            var triple = Triple.of(partitionInfo, row, keyAndStaticScore + rowScore);
```

Comment:
I think it's worth creating a static class to represent this, just make all the fields public final

Reply:
my mistake, I thought the Triple was added in this commit.  go ahead and leave it, we can clean up separately